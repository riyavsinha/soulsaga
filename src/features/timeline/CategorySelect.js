import Divider from '@material-ui/core/Divider';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MenuItem from '@material-ui/core/MenuItem';
import PropTypes from 'prop-types';
import React from 'react';
import TextField from '@material-ui/core/TextField';
import { PRIMARY_CATEGORIES, SECONDARY_CATEGORIES } from './eventiconmaps';

export default function CategorySelect({category, onChange}) {
  return (
    <TextField
      select
      label="Category"
      value={category}
      onChange={onChange}
      className="timeline-category-select__field"
      margin="normal"
      SelectProps={{
        MenuProps: {
          className: 'timeline-category-select__menu',
        },
      }}
      InputProps={{
        classes: {
          input: 'timeline-category-select__input',
        },
      }}
    >
      <MenuItem value="Other">
        <em>Other</em>
      </MenuItem>
      <Divider />
      {Object.keys(PRIMARY_CATEGORIES).map(c => (
        <MenuItem key={c} value={c}>
          <ListItemIcon>{PRIMARY_CATEGORIES[c]}</ListItemIcon>
          <ListItemText inset primary={c} />
        </MenuItem>
      ))}
      <Divider />
      {Object.keys(SECONDARY_CATEGORIES).map(c => (
        <MenuItem key={c} value={c}>
          <ListItemIcon>{SECONDARY_CATEGORIES[c]}</ListItemIcon>
          <ListItemText inset primary={c} />
        </MenuItem>
      ))}
    </TextField>
  );
};

CategorySelect.propTypes = {
  category: PropTypes.oneOf(
    Object.keys(PRIMARY_CATEGORIES)
      .concat(Object.keys(SECONDARY_CATEGORIES))
      .concat(['Other', '']),
  ),
  onChange: PropTypes.func.isRequired,
};