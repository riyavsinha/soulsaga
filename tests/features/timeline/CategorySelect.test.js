import CategorySelect from 'features/timeline/CategorySelect';
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('timeline/CategorySelect', () => {
  let onChangeFn;

  beforeEach(() => {
    onChangeFn = jest.fn();
  });

  describe('snapshot tests', () => {
    it('renders with selection selected', () => {
      const renderedComponent = renderer
        .create(<CategorySelect category="Fitness" onChange={onChangeFn} />)
        .toJSON();

      expect(renderedComponent).toMatchSnapshot();
    });

    it('renders with "other" selected', () => {
      const renderedComponent = renderer
        .create(<CategorySelect category="Other" onChange={onChangeFn} />)
        .toJSON();

      expect(renderedComponent).toMatchSnapshot();
    });
  });

  it('calls onChange when menu item clicked', () => {
    const wrapper = mount(<CategorySelect category="Other" onChange={onChangeFn}/>);

    // Open menu
    wrapper.find('[role="button"]').simulate('click');
    // Click menu option
    wrapper
      .find('[data-value="Fitness"]')
      .first()
      .simulate('click');
    
    expect(onChangeFn.mock.calls.length).toBe(1);
    expect(onChangeFn.mock.calls[0][0].target.value).toBe("Fitness");
  });
});
