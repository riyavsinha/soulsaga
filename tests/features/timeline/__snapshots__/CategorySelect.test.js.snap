// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timeline/CategorySelect snapshot tests renders with "other" selected 1`] = `
<div
  className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 timeline-category-select__field"
>
  <label
    className="MuiFormLabel-root-16 MuiFormLabel-filled-20 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12"
    data-shrink={true}
  >
    Category
  </label>
  <div
    className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31"
    onClick={[Function]}
  >
    <div
      className="MuiSelect-root-23"
    >
      <div
        aria-haspopup="true"
        aria-pressed="false"
        className="MuiSelect-select-24 MuiSelect-selectMenu-27 MuiInputBase-input-53 MuiInput-input-38 timeline-category-select__input"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <em>
          Other
        </em>
      </div>
      <input
        type="hidden"
        value="Other"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-103 MuiSelect-icon-29"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`timeline/CategorySelect snapshot tests renders with selection selected 1`] = `
<div
  className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 timeline-category-select__field"
>
  <label
    className="MuiFormLabel-root-16 MuiFormLabel-filled-20 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12"
    data-shrink={true}
  >
    Category
  </label>
  <div
    className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31"
    onClick={[Function]}
  >
    <div
      className="MuiSelect-root-23"
    >
      <div
        aria-haspopup="true"
        aria-pressed="false"
        className="MuiSelect-select-24 MuiSelect-selectMenu-27 MuiInputBase-input-53 MuiInput-input-38 timeline-category-select__input"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root-60"
        >
          <svg
            className="mdi-icon "
            fill="currentColor"
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M4.22,14.12L3.5,13.41C2.73,12.63 2.73,11.37 3.5,10.59C4.3,9.8 5.56,9.8 6.34,10.59L8.92,13.16L13.16,8.92L10.59,6.34C9.8,5.56 9.8,4.3 10.59,3.5C11.37,2.73 12.63,2.73 13.41,3.5L14.12,4.22L19.78,9.88L20.5,10.59C21.27,11.37 21.27,12.63 20.5,13.41C19.7,14.2 18.44,14.2 17.66,13.41L15.08,10.84L10.84,15.08L13.41,17.66C14.2,18.44 14.2,19.7 13.41,20.5C12.63,21.27 11.37,21.27 10.59,20.5L9.88,19.78L4.22,14.12M3.16,19.42L4.22,18.36L2.81,16.95C2.42,16.56 2.42,15.93 2.81,15.54C3.2,15.15 3.83,15.15 4.22,15.54L8.46,19.78C8.85,20.17 8.85,20.8 8.46,21.19C8.07,21.58 7.44,21.58 7.05,21.19L5.64,19.78L4.58,20.84L3.16,19.42M19.42,3.16L20.84,4.58L19.78,5.64L21.19,7.05C21.58,7.44 21.58,8.07 21.19,8.46C20.8,8.86 20.17,8.86 19.78,8.46L15.54,4.22C15.15,3.83 15.15,3.2 15.54,2.81C15.93,2.42 16.56,2.42 16.95,2.81L18.36,4.22L19.42,3.16Z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root-61 MuiListItemText-inset-62"
        >
          <span
            className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
          >
            Fitness
          </span>
        </div>
      </div>
      <input
        type="hidden"
        value="Fitness"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-103 MuiSelect-icon-29"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
